app/components/chart.tsx

```diff
@@ -31,62 +77,24 @@ export function Chart({
       const tx = Math.min(Math.max(cx, hlw), cw - hlw);
       l.style.transform = `translate(-50%) translate(${tx}px)`;
     }
-  });
-
-  let chartContent: ReactNode;
-  let label: ReactNode;
-
-  if (data) {
-    const timestamp = new Date();
-    const yMax = Math.max(...data);
-    const yMin = Math.min(...data);
-    const yMinPadded = yMin - (yMax - yMin) * 0.15;
-
-    const bars: Array<ReactElement> = [];
-
-    for (let idx = 0; idx < 24; idx++) {
-      const isNumber = typeof data[idx] === 'number';
-      const value = data[idx] ?? yMin;
-      const percentage: number = (value - yMinPadded) / (yMax - yMinPadded);
-      bars[idx] = (
-        <div
-          style={{ transition: 'all .2s' }}
-          className={'flex grow items-end' + (idx == hour ? ' brightness-[1.30]' : '')}
-          onMouseEnter={() => setHour(idx)}
-          key={idx}
-        >
-          <div
-            style={{
-              transition: 'all .2s',
-              backgroundColor: 'hsla(var(--chart))',
-              height: percentage * 100 + '%',
-              filter: isNumber ? '' : 'saturate(0%)',
-            }}
-            className="grow basis-[1px]"
-          ></div>
-        </div>
-      );
-    }
+  }, [hour, refs.container, refs.label]);
 
-    label = Label({ value: data[hour], time: new Date(timestamp.setHours(hour, 0)) });
-    chartContent = (
-      <div
-        className="flex grow cursor-pointer items-stretch pt-[2.5em]"
-        onMouseLeave={() => setHour(new Date().getHours())}
-      >
-        {bars}
-      </div>
-    );
-  } else {
-    chartContent = (
-      <div className="flex grow items-center justify-center">
-        <h3 className="text-[2em]">No Data</h3>
-      </div>
-    );
-  }
+  const [bars, setBars] = useState<Array<ReactElement>>(getBars(null, true));
+
+  // On Data change
+  useEffect(() => {
+    setBars(getBars(data));
+  }, [data, getBars]);
+
+  // Time changes we must update label position
```

Comment:
Maby change already at 30 past?
ex: when the time is ##:30+, does it make more sense to look at the next upcoming hour? (the time is 17:55 of the screenshot)
![image](https://github.com/user-attachments/assets/88cc32b3-e9eb-4dd0-a713-a663c36232e0)